daemon.go

```diff
@@ -177,5 +177,8 @@ func (d *Daemon) Close() error {
 		merr = multierror.Append(merr, err)
 	}
 
-	return multierror.Flatten(merr)
+	if merr != nil {
+		return merr.ErrorOrNil()
+	}
```

Comment:
I don't think we need the `merr != nil` check; the method should work on a nil pointer.

Reply:
so just `return merr.ErrorOrNil()`.

Reply:
My mistake. My IDE gave an error about a potential nil receiver, but the implementation of that method does indeed check for nil receiver